uom_crm_dna.py

```diff
@@ -1,83 +1,235 @@
-import pandas as pd
+"""
+Script to see whether or not variants supplied by known reference standards are called for the CRM assay
+during uncertainty of measurement runs
+"""
+
+
 from glob import glob
 import subprocess
+import sigfig
+import argparse
+from datetime import datetime
+from datetime import date
+import yaml
+import csv
+
+
+###########################
+###	Functions	###
+###########################
+
+def parse_crm_samples(crm_config_list):
+        """
+        Load uncertainty of measurement sample IDs and their corresponding reference standard into a dictionary
+        """
+	# This is the uncertainty of measurment sample ID:reference standard files
+        config_file = open(sample_type, 'r')
+ 
+        config_dict = {}
+
+        for config in crm_config_list:
+
+        		# Loading in the yaml file
+                	sampleid_dict = yaml.load(config_file, Loader=yaml.FullLoader)
+        	
+        		# Convert contents of yaml file into a dictionary
+        		for key in sampleid_dict:
+               		
+               			config_dict[key] = sampleid_dict[key]
+                        
+        return config_dict
+
+
+
+def crm_sample_file_list(reference_dict, sample_path): 
+	"""
+	Get all UOM crm sample filepaths from last year into list
+	"""
+	
+	reference_dict = parse_crm_samples([crm_config_list])
+
+	filepath_list = []
+
+	current_year = datetime.today().year
+	last_year = [str(current_year -1)[2:]]
+	
+	# Getting the sample IDs
+	for sample in reference_dict:
+
+		# Getting all files ran last year
+		for year in last_year:
+
+		# Keep the root path, all files from last year with the sample
+		filepaths += glob(f'/Output/results/{year}*/NGHS-101X/{sample}/{year}*_VariantReport.txt')
+
+		if len(filepaths) == 0:
+
+			print(f'no files for sample {sample}')
+
+		filepath_list = filepath_list + filepaths
+
+	return filepath_list #loops dont work with this command?
+
+	##tsv_files_to_analyse = sample_file_list(reference_dict)
+
+
+
+def get_variants_in_crm_files(filepaths):
+	"""
+	Put position and base change of each sample file into a dictionary
+	"""
+
+	samples_dict = {}
+
+	for sample in filepaths:
+
+		sample_name = sample.split('/')[-1].strip('_VariantReport.txt') #not sure if this will do what i want (just get sample ID)
+
+		samples_dict[sample_name] = {} # MAYBE MOVE THIS TO GET SAMPLES IN SEPARATE DICTIONARIES?
+
+		with open(sample) as file:
+
+			# split the lines by tab
+			file = csv.reader(file, delimeter='\t')
```

Comment:
It's best not to re-use variables, `file` was used when defining the loop on line 90 and then redefined on line 93